### ISSUE TITLE
Is it possible to detect if a tag was rebuilt due to a Dockerfile update?

### ISSUE THREAD

**Issue by smgtkn** [docker-library/faq#34]
Since there are several reasons why official images are rebuild (base image update triggered builds, scheduled builds, Dockerfile update triggered builds), it is hard to understand if a tag was rebuilt due to an update on its Dockerfile. Is there a way to determine whether a tag was rebuilt due to the appearance of a new Dockerfile, pointed to by a new GitCommit, in its library definition file? 
To put it another way, is there a way to determine which of the historical digests for the same tag was built due to a change in its Dockerfile?

**Comment by tianon** [comment]
Your best bet is likely going to be the "history" object in the image config, which is very _roughly_ the original `Dockerfile` (barring multi-stage builds), which is also part of the commit history in https://github.com/docker-library/repo-info (turned roughly back into readable image history).

With our newer builds that include builtkit's provenance, the full original `Dockerfile` is included, but it's a bit buried -- better though, we also include annotations with some of the build metadata: https://explore.ggcr.dev/?image=docker@sha256:6bb6b266d173937b4859871f556ecf8e1d822ea197f821c8b1e784452da13e55&mt=application%2Fvnd.oci.image.manifest.v1%2Bjson&size=3705#:~:text=%22-,annotations,-%22%3A%20%7B (currently only on a small subset of the program's images, but will be rolled out to all of them eventually / over time).

**Comment by tianon** [comment]
For images on our new build system (which will eventually be all of them), this is made slightly better by https://github.com/docker-library/meta-scripts/pull/17: the ["image index"](https://github.com/opencontainers/image-spec/blob/v1.1.0-rc6/image-index.md) will include a `org.opencontainers.image.created` annotation that's set roughly to the actual timestamp of our running `docker buildx build` (where the same annotation inside the ["image manifest"](https://github.com/opencontainers/image-spec/blob/v1.1.0-rc6/manifest.md) will be set to the same `SOURCE_DATE_EPOCH` value as the "created" metadata inside the image config).

See https://explore.ggcr.dev/?image=docker:dind@sha256:91c7180444442d2596355a2859317d86c73025c63c67bceca3e16cd09a08e649 for an example of that.
